nixos/modules/services/web-apps/libretranslate.nix

```
@@ -0,0 +1,197 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.libretranslate;
+  ltmanageKeysCli = pkgs.writeShellScriptBin "ltmanage-keys" ''
+    set -a
+    export HOME="/var/lib/libretranslate"
+    sudo=exec
+    if [[ "$USER" != ${cfg.user} ]]; then
+      sudo='exec /run/wrappers/bin/sudo -u ${cfg.user} --preserve-env'
+    fi
+    $sudo ${cfg.package}/bin/ltmanage keys --api-keys-db-path ${cfg.dataDir}/db/api_keys.db "$@"
+  '';
+
+in {
+  options = {
+    services.libretranslate = {
+      enable = lib.mkEnableOption (lib.mdDoc "LibreTranslate service");
+      package = lib.mkPackageOption pkgs "libretranslate" {
+        default = [ "libretranslate" ];
+        extraDescription = lib.mdDoc "LibreTranslate package to use.";
+      };
+      user = lib.mkOption {
+        type = lib.types.str;
+        default = "libretranslate";
+        description = lib.mdDoc "User account under which libretranslate runs.";
+      };
+      group = lib.mkOption {
+        type = lib.types.str;
+        default = "libretranslate";
+        description = lib.mdDoc "Group account under which libretranslate runs.";
+      };
+      host = lib.mkOption {
+        description = lib.mdDoc "The address the application should listen on.";
+        type = lib.types.str;
+        default = "127.0.0.1";
+      };
+      port = lib.mkOption {
+        type = lib.types.port;
+        default = 5000;
+        description = lib.mdDoc "The the application should listen on.";
+      };
+      dataDir = lib.mkOption {
+        type = lib.types.path;
+        default = "/var/lib/libretranslate";
+        example = "/srv/data/libretranslate";
+        description = lib.mdDoc "The data directory.";
+      };
+      threads = lib.mkOption {
+        type = lib.types.nullOr lib.types.int;
+        default = null;
+        example = 8;
+        description = lib.mdDoc "Set number of threads.";
+      };
+      enableApiKeys = lib.mkOption {
+        type = lib.types.bool;
+        default = false;
+        example = true;
+        description = lib.mdDoc "Enable API keys database.";
+      };
+      disableWebUI = lib.mkOption {
+        type = lib.types.bool;
+        default = false;
+        example = true;
+        description = lib.mdDoc "Disable web ui.";
```

suggestion: just for consistency sake with so many other `enableXYZ`/`disableXYZ` options in nixpkgs.

```suggestion
        description = lib.mdDoc "Whether to enable the API keys database.";
      };
      disableWebUI = lib.mkOption {
        type = lib.types.bool;
        default = false;
        example = true;
        description = lib.mdDoc "Whether to disable the Web UI.";
```

---

Fixed.